GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRTextViewSceneObject.java

```diff
@@ -258,6 +378,168 @@ public void setBackgroundColor(int color) {
     }
 
     /**
+     * set the number of lines for this text
+     * @param lines
+     */
+    public void setNumberLines(int lines) {
+        mTextView.setLines(lines);
+        mIsChanged = true;
+    }
+    /**
+     * Min Lines for this text
+     * @return
+     */
+    public int getMinLines() {
+        return mTextView.getMinLines();
+    }
+
+    /**
+     * max lines for this text
+     * @return
+     */
+    public int getMaxLines() {
+        return mTextView.getMaxLines();
+    }
+
+    /**
+     * Saves the lines of text and used later to calculate the
+     * length in pixels based on font-family, size, etc.
+     * @param textString
+     */
+    public void setTextStrings(String[] textString) {
```

Comment:
The array and the related methods shouldn't be necessary. The text view can just accept one string, detect if it is multiline (look for '\n') and in case it is, handle all the measurements behind the scenes.